src/components/ApplicationCommentsSummary/ApplicationCommentsSummary.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * This file is part of the Digital Planning Register project.
+ *
+ * Digital Planning Register is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Digital Planning Register is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Digital Planning Register. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import {
+  PublicCommentSummary,
+  SpecialistCommentSummary,
+} from "@/types/odp-types/schemas/postSubmissionApplication/data/CommentSummary";
+import { SentimentIcon } from "../SentimentIcon";
+import "./ApplicationCommentsSummary.scss";
+import { Button } from "../button";
+import { capitalizeFirstLetter } from "@/util";
+import { pascalToSentenceCase } from "@/util/pascalToSentenceCase";
+import Link from "next/link";
+
+export interface ApplicationCommentsSummaryProps {
+  type?: "public" | "specialist";
+  summary?: PublicCommentSummary | SpecialistCommentSummary;
+  reference: string;
+  councilSlug: string;
+}
+
+export const renderHeaderContent = (
+  summary: PublicCommentSummary | SpecialistCommentSummary,
+  type: "public" | "specialist",
+) => {
+  if (type === "public") {
+    return (
+      <p className="govuk-heading-m">
+        {summary.totalComments} comment{summary.totalComments !== 1 && "s"}{" "}
+        received
+      </p>
+    );
+  }
+
+  const specialistSummary = summary as SpecialistCommentSummary;
```

Comment:
You're using a type assertion here which is not type-safe because it forces TypeScript to treat summary as a `SpecialistCommentSummary` without actually checking if it is. 

For better type safety use:

```ts
if (type === "specialist" && "totalConsulted" in summary) { ... }
```